tests/suites/cli/display_clouds.sh

```diff
@@ -111,6 +111,90 @@ EOF
 	fi
 }
 
+run_controller_clouds() {
+	echo
+
+	juju add-cloud my-ec2 -f "./tests/suites/cli/clouds/myclouds.yaml" --force --controller ${BOOTSTRAPPED_JUJU_CTRL_NAME}
+	OUT=$(juju clouds --controller ${BOOTSTRAPPED_JUJU_CTRL_NAME} --format=json 2>/dev/null | jq '.[]')
+
+	EXPECTED=$(
+		cat <<'EOF'
+{
+  "defined": "public",
+  "type": "ec2",
+  "auth-types": [
+    "access-key"
+  ],
+  "regions": {
+    "us-west-1": {
+      "endpoint": "https://ec2.us-west-1.amazonaws.com"
+    },
+    "us-west-2": {
+      "endpoint": "https://ec2.us-west-2.amazonaws.com"
+    }
+  },
+  "users": {
+    "admin": {
+      "display-name": "admin",
+      "access": "admin"
+    }
+  }
+}
+EOF
+	)
+	if [[ "${OUT}" != *"${EXPECTED}"* ]]; then
+		echo "expected ${EXPECTED}, got ${OUT}"
+		exit 1
+	fi
+}
+
+run_controller_credentials() {
+	echo
+
+	juju add-credential my-ec2 -f "./tests/suites/cli/clouds/mycreds.yaml" --controller ${BOOTSTRAPPED_JUJU_CTRL_NAME}
+	OUT=$(juju credentials --controller ${BOOTSTRAPPED_JUJU_CTRL_NAME} --format=json 2>/dev/null | jq '.[] ."my-ec2"')
+
+	EXPECTED=$(
+		cat <<'EOF'
```

Comment:
Is this the only way to get a multi-line string in Bash?

Reply:
maybe we could store these expected values in files to keep the test code a bit cleaner? This is what the bundle tests do, for example.

Reply:
yes, this is what bash requires. i'd much rather keep the string in situ so you can very easily see what's being used
i have no issue with this syntax. is there a problem?
the bundle stuff is to specify the input to the command, and we do do that here too to supply the cloud yaml